src/ngCookies/cookieWriter.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+/**
+ * @name $$cookieWriter
+ * @requires $document
+ *
+ * @description
+ * This is a private service for writing cookies
+ *
+ * @param {string} name Cookie name
+ * @param {string=} value Cookie value (if undefined, cookie will be deleted)
+ * @param {Object=} options Object with options that need to be stored for the cookie.
+ */
+function $$CookieWriter($document, $log, $browser) {
+  var cookiePath = $browser.baseHref();
+  var rawDocument = $document[0];
+
+  function buildCookieString(name, value, options) {
+    options = angular.extend({path: cookiePath}, options);
```

Comment:
Rather than use `extend` here we could simply set a `path` variable and use the rest of the `options` object as-is...


```js
var path = angular.isDefined(options.path) ? options.path : cookiePath;
```